### PR TITLE
Document telemetry methods

### PR DESCRIPTION
**Pull request by jswinarton**
Fixes PSP-387

Also, add a missing log message to one event in `check_refunds` that was showing up as an unlabelled event in Sentry.

### CODE REVIEW

File: src/lib/telemetry.rs [hunk 1]
```diff
@@ -135,6 +135,31 @@ pub enum LogKey {
...
+    /// by adding a known suffix to the base value.
+    ///
+    /// Note that the value produced by the method must be a valid Enum value.
+    /// If the stringified value is invalid, the original enum value will be
+    /// returned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// pub fn abstract_logger(base_key: &LogKey) {
```
Comment (birdsarah) [inline src/lib/telemetry.rs]: There's no need to make this a generic example of an "abstract logger".

Just keep the 159 and 160 with what add_suffix actually returns including in the case of an unknown Enum.

File: src/lib/telemetry.rs [hunk 2]
```diff
@@ -306,8 +408,8 @@ pub mod test_telemetry {
 
     #[test]
     fn get_log_key_with_valid_suffix() {
-        let expected = LogKey::TestEnding;
-        let actual = LogKey::Test.add_suffix("ending");
+        let expected = LogKey::TestStarting;
+        let actual = LogKey::Test.add_suffix("starting");
```
Comment (birdsarah) [inline src/lib/telemetry.rs:406]: I'm curious why this change is needed.

File: src/lib/telemetry.rs [hunk 3]
```diff
@@ -228,7 +302,35 @@ macro_rules! info_and_incr {
...
+/// use std::error::Error;
+///
+/// let statsd = StatsD::new(&settings);
+///
+/// error_and_incr!(
+///     statsd,
+///     LogKey::CleanupAicArchiveFailed,
+///     key = "value",
+///     "Some log message"
+/// )
```
Comment (birdsarah) [inline src/lib/telemetry.rs:318]: add a note about the format of the statsd key that gets created e.g. cjms.cleanup-aic-archive-failed

**Review by birdsarah — APPROVED** [review]
Added some requests but approving because they're small so feel free to merge after push.

File: src/lib/telemetry.rs (hunk 2 above)
Comment (jswinarton) [reply src/lib/telemetry.rs:406]: Ah, I meant to comment this. I was thrown at first because "suffix" and "ending" are synonyms, so at first read it appears if there is a relation when there isn't. I just changed it to make it clearer that they are separate things.

File: src/lib/telemetry.rs (hunk 3 above)
Comment (jswinarton) [reply src/lib/telemetry.rs:318]: I added a note about stringification to the `info!` macro. However, the exact format of the key value is dependent on the implementation of both the LogKey enum and the StatsD impl. Since it touches many pieces I don't think it makes sense to stick the docs on the macro itself -- instead I opted to add a paragraph to the README on standard use. Let me know what you think.

File: src/lib/telemetry.rs (hunk 1 above)
Comment (jswinarton) [reply src/lib/telemetry.rs]: done.
